src/ziggurat/init.clj

```diff
@@ -28,38 +27,102 @@
        (mount/with-args args)
        (mount/start))))
 
-(defn start
-  "Starts up Ziggurat's config, reporters, actor fn, rabbitmq connection and then streams, server etc"
-  [actor-start-fn stream-routes actor-routes]
+(defn- start-rabbitmq-connection [args]
+  (start* #{#'messaging-connection/connection} args))
+
+(defn- start-rabbitmq-consumers [args]
+  (start-rabbitmq-connection args)
+  (messaging-consumer/start-subscribers (get args :stream-routes)))
+
+(defn- start-rabbitmq-producers [args]
+  (start-rabbitmq-connection args)
+  (messaging-producer/make-queues (get args :stream-routes)))
+
+(defn start-stream [args]
+  (start-rabbitmq-producers args)
+  (start* #{#'streams/stream} args))
+
+(defn start-management-apis [args]
+  (start-rabbitmq-connection args)
+  (start* #{#'server/server} (dissoc args :actor-routes)))
+
+(defn start-server [args]
+  (start-rabbitmq-connection args)
+  (start* #{#'server/server} args))
+
+(defn start-workers [args]
+  (start-rabbitmq-producers args)
+  (start-rabbitmq-consumers args))
+
+(defn- stop-rabbitmq-connection []
+  (mount/stop #'messaging-connection/connection))
+
+(defn stop-workers []
+  (stop-rabbitmq-connection))
+
+(defn stop-server []
+  (mount/stop #'server/server)
+  (stop-rabbitmq-connection))
+
+(defn stop-stream []
+  (mount/stop #'streams/stream)
+  (stop-rabbitmq-connection))
+
+(defn stop-management-apis []
+  (mount/stop #'server/server)
+  (stop-rabbitmq-connection))
+
+(def valid-modes-fns
```

Comment:
Does vector make sense here or should we make a map with explicit keyword for start fn and stop fn ?